dashboard/test/models/contact_rollups_pardot_memory_test.rb

```diff
@@ -63,29 +63,52 @@ class ContactRollupsPardotMemoryTest < ActiveSupport::TestCase
     end
   end
 
+  test 'download_deleted_pardot_prospects' do
+    deleted_prospects = [
+      {'id' => '1', 'email' => 'earth@rollups.com'},
+      {'id' => '2', 'email' => 'mars@rollups.com'},
+      # There could be multiple prospects with the same email address, but with different pardot_id.
+      {'id' => '3', 'email' => 'earths@rollups.com'}
+    ]
+    PardotV2.stubs(:retrieve_prospects).once.yields(deleted_prospects)
+
+    ContactRollupsPardotMemory.download_deleted_pardot_prospects
+
+    deleted_prospects.each do |prospect|
+      record = ContactRollupsPardotMemory.find_by(email: prospect['email'])
+      refute_nil record
+      refute_nil record.data_rejected_at
+      assert_equal PardotHelpers::ERROR_PROSPECT_DELETED_FROM_PARDOT, record.data_rejected_reason
+    end
```

Comment:
If part of this test is checking that it can handle multiple prospects with the same email, should there be a check that there are 2 records instead of 3? Although I guess that is kind of enforced by the email uniqueness constraint on `ContactRollupsPardotMemory`.

Reply:
Correct, it relies on table constraints to enforce that emails are unique.

Reply:
Btw, I break this verification into multiple assertions instead of combining them all in the `find_by` statement. It will be easier to pinpoint the issue if the test fails.